### DISCUSSION TITLE
AssertionError: size and positions (lon, lat, depth) don't have the same lengths.

### DISCUSSION THREAD

**Discussion by TomMBIO** [Parcels-code/Parcels#1560] [General]
Hi everyone.

I am trying to release particles from a list of lat and long coordinates, but am having trouble getting this to work. I keep encountering this error: "AssertionError: size and positions (lon, lat, depth) don't have the same lengths." I'm not sure why this is happening; my latitude and longitude lists are the same length, and I have specified Depth=None in the model (I'm not using depth for now). Am I missing something?

I am going to attach my code and also the file containing my coordinates. The reason I am releasing particles from specific sites is they represent areas of habitat that my study species presumably disperse from.

Thank you,
Tom

[parcels_testing (1).md](https://github.com/OceanParcels/parcels/files/15293441/parcels_testing.1.md)
[rocky_gridpoints_pruned.csv](https://github.com/OceanParcels/parcels/files/15293451/rocky_gridpoints_pruned.csv)

**Comment by erikvansebille** [comment]
Perhaps not use `depth=None`, but `depth=0` instead? Same for `time`.

Or, if that doesn't work, use `ParticleSet.from_list()`, using `np.linspace` for `lon` and `lat` (like [particleset.from_line under the hood](https://github.com/OceanParcels/parcels/blob/a0ea4044c3ced351b65ec3a19ea3bd8d29f8bd3d/parcels/particleset.py#L528-L531)) and make `depth` a vector of zeros: `depth=[0]*len(particle_lon)`

**Reply by TomMBIO** [reply]
Will give this a shot, thank you!
